### PR TITLE
NAS-125005 / 24.04 / Fix periodic sync of SMB share ACL with datastore

### PR DESCRIPTION
**Pull request by anodos325**
SMB share ACLs may be modified by by SMB clients or locally outside of our APIs by intrepid administrators. Currently we periodically check whether the file has changed behind our back and update stored copies of the SMB share ACL (in our config file). The share ACL stored in DB is base64 encoded security descriptor blob from share_info.tdb.

This edge case did not have adequate test coverage and regressed after an API redesign for the cobia release.

### CODE REVIEW

**Review by mgrimesix — APPROVED** [review]

File: src/middlewared/middlewared/plugins/smb_/sharesec.py [hunk 1]
```diff
@@ -44,15 +44,13 @@ class Config:
         'data_type': 'BYTES'
     }
 
-    @private
```
Comment (bmeagherix) [inline src/middlewared/middlewared/plugins/smb_/sharesec.py:47]: If you are removing private from this (and others), then should some documentation be provided?

TBH, I'd lean more towards keeping these private.  Is there a reason you're changing it?

**Review by bmeagherix — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/smb_/sharesec.py (hunk 1 above)
Comment (anodos325) [reply src/middlewared/middlewared/plugins/smb_/sharesec.py:47]: At some point in the past I made the whole plugin private. These are redundant.

**Review by bmeagherix — APPROVED** [review]
